syntax = "proto3";
package role;


// ================================
// サービス定義
// ================================

service RoleService {
  // Roleの生成   
  rpc CreateRole(CreateRoleRequest) returns (RoleResponse);

  // Roleの取得
  rpc GetRole(GetRoleRequest) returns (RoleResponse);

  // Roleの更新
  rpc UpdateRole(UpdateRoleRequest) returns (RoleResponse);

  // Roleの削除
  rpc DeleteRole(DeleteRoleRequest) returns (DeleteRoleResponse);

  // Roleの一覧取得
  rpc ListRoles(ListRolesRequest) returns (ListRolesResponse);
}

// ================================
// リクエスト / レスポンスメッセージ
// ================================
message CreateRoleRequest {
  string name = 1;
  string description = 2;
}

message GetRoleRequest {
  string id = 1;
}

message UpdateRoleRequest {
  string id = 1;
  string name = 2;
  string description = 3;
}

message RoleResponse{
  Role role = 1;
}

message DeleteRoleRequest {
  string id = 1;
}

message DeleteRoleResponse {
  bool success = 1;
}

message ListRolesRequest {
  int32 page = 1;
  int32 page_size = 2;
}

message ListRolesResponse {
  repeated Role roles = 1;
  int32 total_count = 2;
}


message Role {
    string id = 1;
    string name = 2;
    string description = 3;
}





